src/cockpit/channels/filesystem.py

```diff
@@ -160,6 +161,44 @@ def do_yield_data(self, options: JsonObject) -> Generator[bytes, None, JsonObjec
             raise ChannelError('internal-error', message=str(exc)) from exc
 
 
+class FSReplaceAttrs:
+    uid: 'int | None' = None
+    gid: 'int | None' = None
+    supported_attrs = ['user', 'group']
+
+    def __init__(self, value: JsonObject) -> None:
+        # Any unknown keys throw an error
+        unsupported_attrs = list(value.keys() - self.supported_attrs)
+        if unsupported_attrs:
+            raise ChannelError('protocol-error',
+                               message=f'"attrs" contains unsupported key(s) {unsupported_attrs}',
+                               unsupported_attrs=unsupported_attrs) from None
+
+        user = get_str_or_int(value, 'user', None)
+        group = get_str_or_int(value, 'group', None)
+
+        if user is not None and group is None:
+            raise ChannelError('protocol-error', message='"group" attribute is empty while "user" is provided')
+        if group is not None and user is None:
+            raise ChannelError('protocol-error', message='"user" attribute is empty while "group" is provided')
+
+        if isinstance(user, str):
+            try:
+                self.uid = pwd.getpwnam(user).pw_uid
+            except KeyError:
+                raise ChannelError('internal-error', message=f'uid not found for {user}') from None
```

Comment:
`internal-error` seems wrong to me, this isn't "our" fault. This happens if the user specifies a nonexisting user/group, so perhaps `not-found`?